docs: edit getting started start-forms.md (#39681)
angular/angular: aio/content/start/start-forms.md

# Using forms for user input

This guide builds on the [Managing Data](start/start-data "Try it: Managing Data") step of the Getting Started tutorial, [Get started with a basic Angular app](start "Get started with a basic Angular app").

This section walks you through adding a form-based checkout feature to collect user information as part of checkout.

## Define the checkout form model

This step shows you how to set up the checkout form model in the component class.
The form model is the source of truth for the status of the form.

1. Open `cart.component.ts`.

> 1. Angular's `FormBuilder` service provides convenient methods for generating controls.
>   Import the `FormBuilder` service from the `@angular/forms` package.

Reviewer: I'd reverse this. Put the focus on the user task:

```suggestion
1. Import the `FormBuilder` service from the `@angular/forms` package.
    This service provides convenient methods for generating controls.
  
```

    <code-example header="src/app/cart/cart.component.ts" path="getting-started/src/app/cart/cart.component.ts" region="imports">
    </code-example>

> 1. The `ReactiveFormsModule` provides the `FormBuilder` service, which `AppModule`, in `app.module.ts`, already imports.
>   Inject the `FormBuilder` service in the `CartComponent` `constructor()`.

Reviewer: ```suggestion
1. Inject the `FormBuilder` service in the `CartComponent` `constructor()`.
    This service is part of the `ReactiveFormsModule` module, which you've already imported.
  
```

    <code-example header="src/app/cart/cart.component.ts" path="getting-started/src/app/cart/cart.component.ts" region="checkout-form">
    </code-example>

1. To gather the user's name and address, set the `checkoutForm` property and a form model containing `name` and `address` fields, using the `FormBuilder` `group()` method.

>   Add this between the curly braces, `{}`, of the `constructor()`.

Reviewer: It's best to avoid ambiguous `this` statements. Perhaps:

```suggestion
  Add this property between the curly braces, `{}`, of the constructor.
```

    <code-example header="src/app/cart/cart.component.ts" path="getting-started/src/app/cart/cart.component.ts" region="checkout-form-group"></code-example>

> 1. For the checkout process, users need to submit their name and address.
>   When they submit their order, the form should reset and the cart should clear.

Reviewer: ```suggestion
1. Define an `onSubmit()` method to process the form.
    This method allows users to submit their name and address.
    In addition, this method uses the `clearCart()` method of the `CartService` to reset the form and clear the cart.
```
Again, in a procedure, it's best to focus on the user task first, and then follow up with an explanation.

  Define an `onSubmit()` method to process the form.

>   Use the `CartService` `clearCart()` method to empty the cart items and reset the form after its submission.

Reviewer: If you go with my previous comment, you should be able to delete these two lines.

  The entire cart component class is as follows:

  <code-example header="src/app/cart/cart.component.ts" path="getting-started/src/app/cart/cart.component.ts">
  </code-example>

## Create the checkout form

Use the following steps to add a checkout form at the bottom of the Cart view.

1. At the bottom of `cart.component.html`, add an HTML `<form>` element and a **Purchase** button.

>   Use a `formGroup` property binding to bind `checkoutForm` to the HTML `<form>`.

Reviewer: ```suggestion
1. Use a `formGroup` property binding to bind `checkoutForm` to the HTML `<form>`.
```
Each user task should be a step.

  <code-example path="getting-started/src/app/cart/cart.component.html" header="src/app/cart/cart.component.html (cart component template detail)" region="checkout-form-1">
  </code-example>

> 1. Add `<input>` fields for `name` and `address` each with a`formControlName` attribute binding to the `checkoutForm` form controls for `name` and `address` to their `<input>` fields.

Reviewer: ```suggestion
1. Add `<input>` fields for `name` and `address`, each with a`formControlName` attribute that binds to the `checkoutForm` form controls for `name` and `address` to their `<input>` fields.
```

  <code-example path="getting-started/src/app/cart/cart.component.html" header="src/app/cart/cart.component.html (cart component template detail)" region="checkout-form-1">
  </code-example>

1. Add `<input>` fields for `name` and `address`.

> Use the `formControlName` attribute binding to bind the `checkoutForm` form controls for `name` and `address` to their `<input>` fields.

Reviewer: ```suggestion
1. Use the `formControlName` attribute binding to bind the `checkoutForm` form controls for `name` and `address` to their `<input>` fields.
```

Each user task should be as tep.

Author: Update: When walking through the steps, I needed to put these two back together as creating the inputs and adding the `formControlName` attributes is actually one step that the developer takes simultaneously on the inputs. You are creating the inputs with the attributes as part of those inputs rather than a separate entities. I had separated them according to your comment here but when I walked through it, it made sense to have these two together. If you still want to separate these out, we can go that way too, but I think when you see the way that it's written and try to follow the steps it'll make sense this way.
